core/dbt/contracts/graph/unparsed.py

```diff
@@ -429,13 +431,34 @@ class UnparsedExposure(dbtClassMixin, Replaceable):
     type: ExposureType
     owner: ExposureOwner
     description: str = ""
+    label: Optional[str] = None
     maturity: Optional[MaturityType] = None
     meta: Dict[str, Any] = field(default_factory=dict)
     tags: List[str] = field(default_factory=list)
     url: Optional[str] = None
     depends_on: List[str] = field(default_factory=list)
     config: Dict[str, Any] = field(default_factory=dict)
 
+    @classmethod
+    def validate(cls, data):
+        super(UnparsedExposure, cls).validate(data)
+        # TODO: what do we actually want to rules to be?
+        if "name" in data:
+            errors = []
+            if " " in data["name"]:
+                errors.append("cannot contain spaces")
+            if len(data["name"]) > 126:
+                errors.append("cannot contain more than 126 characters")
+            if not (re.match(r"^[A-Za-z]", data["name"])):
+                errors.append("must begin with a letter")
+            if not (re.match(r"[\w-]+$", data["name"])):
+                errors.append("must contain only letters, numbers and underscores")
+
```

Comment:
@jtcohen6 this is just a copy of the metrics name validation.  It's overkill here I believe.  The issue wasn't super clean on what to validate on so I just copied it over to get something up.  Let me know what rules should apply!